python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py

```diff
@@ -0,0 +1,582 @@
+# quantifying the variability of the observed phenomenon.
+#
+# 1.1. Likelihood of the linear regression model
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# The above linear regression model can thus be written:
+#
+# .. math::
+#
+#    \vect{Y} | \vect{\theta}, \tau
+#    {\sim} \mathcal N_{n}(\mat{X} \vect{\theta} ; \tau^{-1} \mat{I}_n +\mat{Q}^{-1}),
+#
+# We then have the following likelihood:
+#
+# .. math::
+#
+#    \mathcal L(\vect{Y}|\vect{\theta}, \tau) =
+#    (2\pi)^{-\frac{n}{2}}
+#    {\rm det}(\tau^{-1} \mat{I}_n+\mat{Q}^{-1})^{-\frac{1}{2}}
+#    e^{-\frac{1}{2}
+#    ||\vect{Y} - \mat{X} \vect{\theta} ||_{\tau^{-1} \mat{I}_n +\mat{Q}^{-1}}^2
+#    }
+#
+# where :math:`||\vect{X}-\vect{Y}||_{\mat{V}}^2` is the Mahalanobis distance between
+# :math:`\vect{X}` and :math:`\vect{Y}` with covariance matrice :math:`\mat{V}`:
+#
+# .. math::
+#
+#    ||\vect{X}-\vect{Y}||_{\mat{V}}^2 = (\vect{X}-\vect{Y})^\top\ \mat{V}^{-1}\ (\vect{X}-\vect{Y}).
+#
+# 1.2. Interval Censorship
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We now assume that, instead of observing directly the
+# :math:`Y_i, i=1,\ldots,n` as described above, we only have access to
+# discretized values :math:`Y_i^{obs} \in \delta \times \mathbb N`, where
+# :math:`\delta > 0` is a grid length and :math:`Y_i^{obs} = \delta \times k` means that
+# :math:`Y_i \in \delta \times \left[ k - \frac{1}{2}; k + \frac{1}{2} \right[`.
+#
+# 1.3. Remarks
+# ~~~~~~~~~~~~
+#
+# -  The presence of a composite matrix :math:`\tau^{-1} \mat{I}_n+\mat{Q}^{-1}` makes
+#    estimation more complex than with a spherical one
+#    :math:`\tau^{-1}\mat{I}_n,` since we would then have explicit (closed-form)
+#    maximum likelihood estimators, and also conjugate priors leading to
+#    explicit full joint posterior distributions.
+# -  Another difficulty is the presence of censored data, since the
+#    likelihood is no more available in closed-form. As we will see, this
+#    can be overcome thanks to Bayesian inference.
+# -  Heteroscedastic linear modeling under interval censorship as
+#    formulated above was originally motivated by an industrial case-study
+#    in seismology, wherein the :math:`Y_i` correspond to the observed
+#    intensity of an earthquake in a distant site, and explanatory
+#    variables :math:`\vect{X}_{i}` are derived from the epicentral distance to the
+#    earthquake's source as well as its characteristics (magnitude,
+#    depth). But it can also arise in many different contexts, as soon as
+#    observations are available with known statistical precisions (hence
+#    the heteroscedasticity) and limited numerical accuracy (hence the
+#    censorship).
+
+# %%
+# 1.4 Simulate the dataset
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+import openturns as ot
+from openturns.viewer import View
+
+ot.Log.Show(ot.Log.NONE)
+ot.RandomGenerator.SetSeed(11123)
+import numpy as np
+
+n = 10
+delta = 0.5
+
+
+X = ot.ComposedDistribution([ot.Dirac(1.0), 10.0 * ot.Normal()]).getSample(n)
```

Comment:
Setting X as the sample from a joint (Dirac, Normal) distribution deserves a comment.

Reply:
In addition to this, the column of ones should explicitly be created as such rather through an artificial Dirac distribution.